src/dlgPackageExporter.cpp

```diff
@@ -662,228 +665,231 @@ void dlgPackageExporter::slot_export_package()
-                               << itDir.fileName()
-                               << "it is a Symlink - avoided to prevent file-system loops!";
-                    continue;
-                }
+        if (entryInfo.isSymLink()) {
+            qWarning() << "dlgPackageExporter::slot_export_package() skipping file: "
+                       << itDir.fileName()
+                       << "it is a Symlink - avoided to prevent file-system loops!";
+            continue;
+        }
 
-                QString nameInArchive = itDir.filePath();
-                nameInArchive.remove(QStringLiteral("%1/").arg(StagingDirName));
+        QString nameInArchive = itDir.filePath();
+        nameInArchive.remove(QStringLiteral("%1/").arg(stagingDirName));
 
-                if       (entryInfo.isDir()) {
-                    directoryEntries.append(nameInArchive);
-                } else if(entryInfo.isFile()) {
-                    fileEntries.insert(nameInArchive, itDir.filePath());
-                }
-            }
+        if       (entryInfo.isDir()) {
+            directoryEntries.append(nameInArchive);
+        } else if(entryInfo.isFile()) {
+            fileEntries.insert(nameInArchive, itDir.filePath());
+        }
+    }
 
 #if defined(Q_OS_WIN32)
-            qt_ntfs_permission_lookup--;
+    qt_ntfs_permission_lookup--;
 #endif
 
-            if (directoryEntries.count() > 1) {
-                std::sort(directoryEntries.begin(), directoryEntries.end());
-            }
+    if (directoryEntries.count() > 1) {
+        std::sort(directoryEntries.begin(), directoryEntries.end());
+    }
 
-            QStringListIterator itDirectoryName(directoryEntries);
-            while (itDirectoryName.hasNext() && isOk) {
-                QString directoryName = itDirectoryName.next();
-                // zip_dir_add(...) returns the index of the
-                // added directory item in the archive or -1 on error:
-                if (zip_dir_add(archive, directoryName.toStdString().c_str(), ZIP_FL_ENC_UTF_8) == -1) {
-                    displayResultMessage(tr("Failed to add directory \"%1\" to package. Error is: \"%2\".")
-                                         .arg(directoryName, zip_strerror(archive)), false);
-                    zip_close(archive);
-                    isOk = false;
-                }
-            }
+    QStringListIterator itDirectoryName(directoryEntries);
+    while (itDirectoryName.hasNext() && isOk) {
+        QString directoryName = itDirectoryName.next();
+        // zip_dir_add(...) returns the index of the
+        // added directory item in the archive or -1 on error:
+        if (zip_dir_add(archive, directoryName.toStdString().c_str(), ZIP_FL_ENC_UTF_8) == -1) {
+            QString errorMsg = tr("Failed to add directory \"%1\" to package. Error is: \"%2\".")
+                                 .arg(directoryName, zip_strerror(archive));
+            zip_close(archive);
```

Comment:
Yeah, didn't make this clear previously - this needed to be the other one that doesn't try to write to the archive file if it already exists:
```suggestion
            zip_discard(archive);
```